connector/connector.py

```diff
@@ -316,6 +326,29 @@ def get_connector_unit(self, base_class):
         return self.backend.get_class(base_class, self.session,
                                       self.model_name)(self)
 
+    @classmethod
+    def create_environment(cls, backend_record, session, model,
+                           connector_env=None):
```

Comment:
When a `connector_env` is given, we already know the `backend_record`, the `session` and the `model` (which may change) so we can wonder if they could not all be optional (but mandatory when we have no `connector_env`). Still, thinking about it, it would make the body of the class method and the understanding of the API more complicated so it maybe doesn't worth the gain of removing the repetition.